### PR TITLE
fix bugs explain with non-correlated query

### LINKED ISSUE #5265: Some In/Exists Subqueries will generate wrong PhysicalPlan
**Describe the bug**

Datafusion can't execute non-correlated subquery now, like in/exists.
So these queries should return `NotImplemented("Physical plan does not support logical expression In/Exists` error. But currently the filter will be pushdown to the TableScan.
```sql
> explain select * from t1 where exists(select 1 from t2 where t2.t2_id > 0);
+---------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
| plan_type     | plan                                                                                                                                                        |
+---------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
| logical_plan  | Projection: t1.t1_id, t1.t1_name, t1.t1_int                                                                                                                 |
|               |   TableScan: t1 projection=[t1_id, t1_name, t1_int], full_filters=[EXISTS (<subquery>)]                                                                     |
|               |     Subquery:                                                                                                                                               |
|               |       Projection: Int64(1)                                                                                                                                  |
|               |         Filter: CAST(t2.t2_id AS Int64) > Int64(0)                                                                                                          |
|               |           TableScan: t2                                                                                                                                     |
| physical_plan | ProjectionExec: expr=[t1_id@0 as t1_id, t1_name@1 as t1_name, t1_int@2 as t1_int]                                                                           |
|               |   CsvExec: files={1 group: [[home/work/tools/datafusion-test-data/join-context/t1.csv]]}, has_header=false, limit=None, projection=[t1_id, t1_name, t1_int] |
|               |                                                                                                                                                             |
+---------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

For the above query, the subquery filter will be  lost in the final `physical_plan`.

**To Reproduce**
As above.

**Expected behavior**
The query should return `NotImplemented("Physical plan does not support logical expression In/Exists` error.

**Additional context**

### PR DESCRIPTION
**Pull request by Lordworms**
## Which issue does this PR close?

Closes #5265

### CODE REVIEW

File: datafusion/core/src/physical_planner.rs [hunk 1]
```diff
@@ -1792,11 +1792,19 @@ impl DefaultPhysicalPlanner {
                             Err(e) => return Err(e),
                         }
                     }
-                    Err(e) => stringified_plans
-                        .push(StringifiedPlan::new(InitialPhysicalPlan, e.to_string())),
+                    Err(err) => {
+                        return Ok(Some(Arc::new(ExplainExec::new(
+                            SchemaRef::new(Schema::new(vec![arrow_schema::Field::new(
```
Comment (alamb) [inline datafusion/core/src/physical_planner.rs:1797]: I think this change means that the explain will not have the logical plan, but instead will have only the error message for creating the physical plan as the "Final Logical Plan"

**Comment by alamb** [comment]
I merged up from main to try and get the CI passing (I won't force push this PR again!)

File: datafusion/core/src/physical_planner.rs [hunk 2]
```diff
@@ -1797,11 +1797,19 @@ impl DefaultPhysicalPlanner {
                             Err(e) => return Err(e),
                         }
                     }
-                    Err(e) => stringified_plans
-                        .push(StringifiedPlan::new(InitialPhysicalPlan, e.to_string())),
```
Comment (alamb) [inline datafusion/core/src/physical_planner.rs:1801]: I played around with this some more this morning -- I think the reason the error isn't displayed is that `InitialPhysicalPlan` is not shown by default in explain plans.

**Review by alamb — COMMENTED** [review]
Thank you @Lordworms  -- I played around with this and I think I found another way to do it:

https://github.com/Lordworms/arrow-datafusion/pull/2

Let me know what you think!

**Comment by Lordworms** [comment]
> Thank you @Lordworms -- I played around with this and I think I found another way to do it:
> 
> [Lordworms#2](https://github.com/Lordworms/arrow-datafusion/pull/2)
> 
> Let me know what you think!

Thanks a lot!

**Review by alamb — APPROVED** [review]
Thank you @Lordworms  🚀

**Pull request merged**